portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,13 +37,28 @@
   
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> comments = new ArrayList<>();
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-    PreparedQuery results = this.datastore.prepare(query);
-    for (Entity entity : results.asIterable()) {
-      comments.add((String) entity.getProperty("content"));
+    int limit = 3;
+    String ERR_MSG = "Comment limit must be a non-negative integer.";
+    try {
+      limit = Integer.parseInt(request.getParameter("limit"));
+    }
+    catch(NumberFormatException e){
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, ERR_MSG);
+      return;
+    }
+    if (limit<0){
```

Comment:
It's also a good idea in general to set a maximum number of comments.

Right now I can crash your server by requesting more comments than your server can hold in memory (assuming there are actually that many in the database)

Reply:
Set max number of comments to return (50).